### ISSUE TITLE
[Widget] Allow different DOM implementations than Phosphor

### ISSUE THREAD

**Issue by Korijn** [flexxui/flexx#418]
I would love to be able to bring-my-own DOM implementation for a Widget. E.g. give me a node that the widget should hook up to and let me generate the DOM using my own means.

It would be awesome to e.g. just attach my own HTML to a node or even leverage something like react.

**Comment by almarklein** [comment]
Some sketches: https://gist.github.com/almarklein/2c4a37a6689fd55b41aa33f8957877c7

**Comment by almarklein** [comment]
This now works:
```py
class MyReactlikeWidget(ui.Widget):

    name = event.StringProp('john', settable=True)
    
    def _render_dom(self):
        return [ui.create_element('span', {}, 'greet: '),
                ui.create_element('b',
                                  dict(onclick=lambda:print('clicked!')),
                                  'hi ' + self.name),
                ui.create_element('span', {},
                                  '<br><button onclick="alert(\'told you!\')">dont click me</button>'),
                ui.create_element('button',
                                  dict(onclick=lambda:window.alert('told you')),
                                  'dont click me'),
                ]
```

**Comment by almarklein** [comment]
Closed by #408
